velox/vector/arrow/Bridge.cpp

```diff
@@ -1112,19 +1114,24 @@ void exportConstantValue(
     // If this is a scalar type, then ConstantVector does not have a vector
     // inside. Wrap the single value in a flat vector with a single element to
     // export it to an ArrowArray.
-    size_t bufferSize = (vec.type()->isVarchar() || vec.type()->isVarbinary())
-        ? sizeof(StringView)
-        : vec.type()->cppSizeInBytes();
+    if (vec.typeKind() == TypeKind::UNKNOWN) {
+      valuesVector = BaseVector::create(vec.type(), 1, pool);
+      valuesVector->setNull(0, true);
```

Comment:
Is `setNull` needed, because `BaseVector::createInternal` would allocate the null buffer as all-null?

```
case TypeKind::UNKNOWN: {
      BufferPtr nulls = allocateNulls(size, pool, bits::kNull);
      return std::make_shared<FlatVector<UnknownValue>>(
          pool,
          UNKNOWN(),
          std::move(nulls),
          size,
          nullptr,
          std::vector<BufferPtr>());
    }
```